### PR TITLE
vimPlugins.mini-*

### PR DESCRIPTION
**Pull request by josh**
## Description of changes

Adds more Neovim subplugins from [echasnovski's mini collection](https://github.com/echasnovski?tab=repositories&q=mini.&type=&language=&sort=).

I used `vim-plugins-updater add` to insert these entries. <del>Should I also run `vim-plugins-updater` itself to generate the `generated.nix` change too? I wasn't sure if that was going to attempt to update all plugin revs.</del>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/vim/plugins/vim-plugin-names [hunk 1]
```diff
@@ -520,7 +520,20 @@ https://github.com/lsig/messenger.nvim/,HEAD,
...
+https://github.com/echasnovski/mini.diff/,HEAD,
+https://github.com/echasnovski/mini.files/,HEAD,
+https://github.com/echasnovski/mini.hipatterns/,HEAD,
+https://github.com/echasnovski/mini.icons/,HEAD,
+https://github.com/echasnovski/mini.indentscope/,HEAD,
+https://github.com/echasnovski/mini.move/,HEAD,
 https://github.com/echasnovski/mini.nvim/,,
+https://github.com/echasnovski/mini.pairs/,HEAD,
+https://github.com/echasnovski/mini.starter/,HEAD,
+https://github.com/echasnovski/mini.surround/,HEAD,
```
Comment (PerchunPak) [inline pkgs/applications/editors/vim/plugins/vim-plugin-names:536]: There are over 40 `mini.*` plugins, why not add all of them then?

image

Comment (josh) [reply pkgs/applications/editors/vim/plugins/vim-plugin-names:560]: ✅ Done!

**Comment by josh** [comment]
@PerchunPak thanks for you help on PR and my other vimplugin related one! Had some other general questions. Is there a way to run `vim-plugins-updater` against a specific plugin name? Would be nice to avoid needing to update all and try to pick out the relevant changes. Also, when do all vim plugins get updated in general? Does it happen automatically on some schedule? Or does someone usually submit a PR that just updates all plugins?

**Comment by PerchunPak** [comment]
> thanks for your help on this PR and also #331008! 

No problem!

> Is there a way to run vim-plugins-updater against a specific plugin name?

Unfortunately, not yet. I am working on a solution (tracking issue: #218584). For now, you can create your own `vim-plugin-names` file and ask the script to generate updates only for that index file (with `-I`, see `--help`). Then you can copy the output to `generated.nix` (it's probably ok if you copy it at the bottom, it will be sorted at the next global update)

> Also, when do all vim plugins get updated in general? Does it happen automatically on some schedule? Or does someone usually submit a PR that just updates all plugins?

Generally once a month, it happens by hand (probably because there may be problems with some patches) when someone notices that some plugin is outdated.

**Comment by PerchunPak** [comment]
Result of `nixpkgs-review pr 331156` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>43 packages built:</summary>
  <ul>
    <li>vimPlugins.mini-ai</li>
    <li>vimPlugins.mini-align</li>
    <li>vimPlugins.mini-animate</li>
    <li>vimPlugins.mini-base16</li>
    <li>vimPlugins.mini-basics</li>
    <li>vimPlugins.mini-bracketed</li>
    <li>vimPlugins.mini-bufremove</li>
    <li>vimPlugins.mini-clue</li>
    <li>vimPlugins.mini-colors</li>
    <li>vimPlugins.mini-comment</li>
    <li>vimPlugins.mini-completion</li>
    <li>vimPlugins.mini-cursorword</li>
    <li>vimPlugins.mini-deps</li>
    <li>vimPlugins.mini-diff</li>
    <li>vimPlugins.mini-doc</li>
    <li>vimPlugins.mini-extra</li>
    <li>vimPlugins.mini-files</li>
    <li>vimPlugins.mini-fuzzy</li>
    <li>vimPlugins.mini-git</li>
    <li>vimPlugins.mini-hipatterns</li>
    <li>vimPlugins.mini-hues</li>
    <li>vimPlugins.mini-icons</li>
    <li>vimPlugins.mini-indentscope</li>
    <li>vimPlugins.mini-jump</li>
    <li>vimPlugins.mini-jump2d</li>
    <li>vimPlugins.mini-map</li>
    <li>vimPlugins.mini-misc</li>
    <li>vimPlugins.mini-move</li>
    <li>vimPlugins.mini-notify</li>
    <li>vimPlugins.mini-nvim</li>
    <li>vimPlugins.mini-operators</li>
    <li>vimPlugins.mini-pairs</li>
    <li>vimPlugins.mini-pick</li>
    <li>vimPlugins.mini-sessions</li>
    <li>vimPlugins.mini-splitjoin</li>
    <li>vimPlugins.mini-starter</li>
    <li>vimPlugins.mini-statusline</li>
    <li>vimPlugins.mini-surround</li>
    <li>vimPlugins.mini-tabline</li>
    <li>vimPlugins.mini-test</li>
    <li>vimPlugins.mini-trailspace</li>
    <li>vimPlugins.mini-visits</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

File: pkgs/applications/editors/vim/plugins/generated.nix [hunk 2]
```diff
@@ -6228,18 +6228,510 @@ final: prev:
...
     pname = "mini.nvim";
-    version = "2024-08-04";
+    version = "2024-08-18";
     src = fetchFromGitHub {
       owner = "echasnovski";
       repo = "mini.nvim";
-      rev = "af673d8523c5c2c5ff0a53b1e42a296ca358dcc7";
-      sha256 = "0rky24war90avndgm9fji0f20d24bgyc3s1spgw5lw443wh9vizl";
+      rev = "d6e6a6fb010cc452b491704aa9ecc96a8aa2cde1";
+      sha256 = "1kxav8z9c40avjqv6npddbvvmjbhn4h291x0qhls9v6ai145spgn";
```
Comment (PerchunPak) [inline pkgs/applications/editors/vim/plugins/generated.nix:6586]: Looks like you accidentally updated main plugin, but I don't think this is an issue

**Review by PerchunPak — APPROVED** [review]
LGTM! I also manually verified there are all plugins from https://github.com/echasnovski?tab=repositories&q=mini&type=&language=&sort=name in `vim-plugin-names` and output of `nixpkgs-review`.

Though I have concerns about the maintainability of such an approach, we will forget about adding new plugins (if that arises) to this list. @teto, is this a problem, or could we merge this?

**Review by GaetanLepage — APPROVED** [review]
LGTM

**Pull request merged**